File: tools/bench/answers/go_haiku/s2/07_matrix_diag.go
package main

import (
	"fmt"
)

func main() {
	matrix := [][]int{
		{5, 1, 2},
		{3, 6, 4},
		{7, 8, 9},
	}
	sum := 0
	for i := 0; i < len(matrix); i++ {
		sum += matrix[i][i]
	}
	fmt.Println(sum)
}

File: tools/bench/answers/rust_haiku/s3/07_matrix_diag.rs
fn main() {
    let matrix = [
        [5, 1, 2],
        [3, 6, 4],
        [7, 8, 9],
    ];

    let mut sum = 0;
    for i in 0..3 {
        sum += matrix[i][i];
    }

    println!("{}", sum);
}
